### PR TITLE
Initial profile model

### PR DESCRIPTION
**Pull request by rizperdana**
### What this PR does (required):
- This MR will add profile model on backend

### Any information needed to test this feature (required):
No test available at the moment

### Any issues with the current functionality (optional):
- No issue so far

### CODE REVIEW

File: server/models/Profile.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const profileSchema = new mongoose.Schema({
+    firstname: { type: String },
+    lastname: { type: String },
+    description: { type: String },
+    availability: { type: [timeSlot] },
+    address: { type: String },
+    dateOfBirth: { type: Date },
+    photo: { type: String },
+    socialMedia: { type: String }
+});
```
Comment (moffatethan) [inline server/models/Profile.js]: Could we add some more formatting here. 

```js
const profileSchema = new mongoose.Schema({
    firstname: { 
        type: String 
    },
    . . . 
});
```

File: server/models/Profile.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require("mongoose");
+const timeSlot = new mongoose.Schema({ start: Date, end: Date })
+
+const profileSchema = new mongoose.Schema({
+    firstname: { type: String },
```
Comment (moffatethan) [inline server/models/Profile.js]: Would like some validation and sanitization on these fields. `required`, `trim`, etc.

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require("mongoose");
+const timeSlot = new mongoose.Schema({ start: Date, end: Date })
```
Comment (moffatethan) [inline server/models/Profile.js]: Not sure if this is the best approach here. We'll likely keep availability pretty basic just monday to friday between X and Y time. Date might not work the best in this case but we'll discuss this more when we work on the feature.

Comment (moffatethan) [reply server/models/Profile.js]: Would be nice to ensure the end time is not larger than the start time too but not a huge deal for me right now. As mentioned before, when we dive into availability we'll spend more time on this.

File: server/models/Profile.js (hunk 2 above)
Comment (moffatethan) [reply server/models/Profile.js]: Change throughout the schema please 👍🏻

**Review by moffatethan — CHANGES_REQUESTED** [review]
Overall decent, let's be mindful of the formatting of our files. Keep things crammed can be hard to review. Make sure we have good validation on our properties is important too.

File: server/models/Profile.js (hunk 3 above)
Comment (rizperdana) [reply server/models/Profile.js]: Alright, gonna make this as independent column for better usage in future

File: server/models/Profile.js (hunk 2 above)
Comment (rizperdana) [reply server/models/Profile.js]: You right, i will put sanitazion and requirement here

File: server/models/Profile.js (hunk 1 above)
Comment (rizperdana) [reply server/models/Profile.js]: Got it, alongside additional sanitazion i will fix the formatting

**Review by moffatethan — COMMENTED** [review]
Good work! @rizperdana Could you invite `moffatethan`, `bonnieli`, `rajivtitus`, and `ugur-akin` to your repository as well please. Thanks!

**Pull request closed without merging**